contracts/airdrop/CruzoAirdrop.sol

```diff
@@ -0,0 +1,100 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/IERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/utils/ERC1155HolderUpgradeable.sol";
+
+import "../transfer-proxy/ITransferProxy.sol";
+
+contract CruzoAirdrop is
+    Initializable,
+    UUPSUpgradeable,
+    OwnableUpgradeable,
+    ERC1155HolderUpgradeable
+{
+    using CountersUpgradeable for CountersUpgradeable.Counter;
+
+    struct Drop {
+        address tokenAddress;
+        uint256 tokenId;
+        uint256 amount;
+        uint256 claimed;
+        mapping(address => bool) claimers;
+    }
+
+    event DropCreated(
+        uint256 id,
+        address tokenAddress,
+        uint256 tokenId,
+        address creator,
+        uint256 amount
+    );
+
+    event DropClaimed(uint256 id, address claimer);
+
+    CountersUpgradeable.Counter private ids;
+
+    ITransferProxy public transferProxy;
+
+    mapping(uint256 => Drop) public drops;
+
+    constructor() {}
+
+    function _authorizeUpgrade(address) internal override onlyOwner {}
+
+    function initialize(ITransferProxy _transferProxy) public initializer {
+        __Ownable_init();
+        transferProxy = _transferProxy;
+    }
+
+    function claim(uint256 _id) external {
+        Drop storage drop = drops[_id];
+        require(drop.amount > 0, "Airdrop: not found");
```

Comment:
In new contracts it is better to implement custom errors instead of require

Reply:
Good point 👍
d5c8ed4eb3598db7ee08befb4c804297030f4376:
![image](https://user-images.githubusercontent.com/20911878/207409044-ea83a8b5-d920-4794-969d-d1366f8051de.png)
ebaa6a9b356af2a69e1c3d828f152fcd57fa60a6:
![image](https://user-images.githubusercontent.com/20911878/207409126-f560be4a-7d04-4fdf-868b-08a774e9e5c1.png)

I'm gonna use custom errors in CruzoGift and CruzoMarket as well